djautotask/api_rest.py

```diff
@@ -223,32 +266,42 @@ def get_headers(self):
         return headers
 
     def build_query_string(self, **kwargs):
-        filter_array = self._build_filter(**kwargs)
+        filter_array = self._build_filters(**kwargs)
         query_obj = {'filter': filter_array}
         self.QUERYSTR = json.dumps(query_obj)
 
-    def _build_filter(self, **kwargs):
+    def _build_filters(self, **kwargs):
         filter_array = []
 
         if 'conditions' in kwargs:
 
-            for condition in kwargs['conditions']:
-
-                if type(condition[0]) == dict:
-                    sub_kwargs = {'conditions': condition[0]['operands']}
-                    filter_obj = {
-                        "op": condition[0]['op'],
-                        "items": self._build_filter(**sub_kwargs)
-                    }
-                    filter_array.append(filter_obj)
-                else:
-                    filter_obj = self._build_filter_obj(*condition)
-                    filter_array.append(filter_obj)
+            conditions = kwargs['conditions']
+
+            if hasattr(conditions, 'operands'):
+                filter_obj = {
+                    "op": conditions.operator,
+                    "items": [
+                        self._build_filter_unit(c)
+                        for c in conditions.operands
+                    ]
+                }
```

Comment:
Since you designed a specific class to handle API conditions, the logic for building those conditions should be put in that class as well. 

The way it is currently is so highly coupled its hard to tell the reason for the separate class at all. As the `AutotaskAPIClient` class has to check the `ApiCondition` instance if it has a specific attribute, and then builds the condition itself. There are even helper methods in `AutotaskAPIClient` for building filters rather than in `ApiCondition`.